### PR TITLE
doc: module and submodule documentation

### PR DESCRIPTION
**Pull request by rubenVB01**
## Description

README files about the Encryption at Rest module and the AWS KMS submodule.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,16 +1,11 @@
-# {Module-Name} Terraform Module
+# Encryption At Rest Terraform Module
 
-Description of the module
+This Terraform module sets up [encryption at rest](https://www.mongodb.com/docs/atlas/security-kms-encryption/) using Customer Key Management for your [MongoDB Atlas](https://www.mongodb.com/products/platform/atlas-database) Project and consists of an [AWS KMS Key](https://github.com/terraform-mongodbatlas-modules/terraform-mongodbatlas-encryption-at-rest/tree/main/modules/aws-kms).
```
Comment (kanchana-mongodb) [inline README.md:3]: Could this be simplified to something similar to the following: 
```suggestion
This Terraform module sets up [encryption at rest](https://www.mongodb.com/docs/atlas/security-kms-encryption/) using your [AWS KMS Key](https://github.com/terraform-mongodbatlas-modules/terraform-mongodbatlas-encryption-at-rest/tree/main/modules/aws-kms) for your [MongoDB Atlas](https://www.mongodb.com/products/platform/atlas-database) Project.
```

File: modules/aws-kms/README.md [hunk 2]
```diff
@@ -1,16 +1,49 @@
...
-Description of the module
+This Terraform submodule sets up [encryption at rest](https://www.mongodb.com/docs/atlas/security-kms-encryption/) using an AWS KMS Key for your [MongoDB Atlas](https://www.mongodb.com/products/platform/atlas-database) Project.
+
+It creates the following resources:
+
+- A MongoDB Atlas Cloud Provider Access Setup.
+- An AWS IAM Role.
+- An AWS IAM Role Policy.
+- A MongoDB Atlas Cloud Provider Access Authorization.
+- Enables Encryption at Rest in the Atlas project.
```
Comment (kanchana-mongodb) [inline modules/aws-kms/README.md]:
```suggestion
It creates the resources to perform the following actions:

- Set up Access to Cloud Provider for MongoDB Atlas.
- Create an AWS IAM Role.
- Create an AWS IAM Role Policy.
- Authorize MongoDB Atlas for Cloud Provider Access.
- Enable Encryption at Rest in the Atlas project.
```

File: modules/aws-kms/README.md [hunk 3]
```diff
@@ -1,16 +1,49 @@
...
+|------|------|
+| [aws_kms_key](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/kms_key) | data source |
+| [mongodbatlas_encryption_at_rest](https://registry.terraform.io/providers/mongodb/mongodbatlas/1.17.6/docs/resources/encryption_at_rest) | resource |
+| [mongodbatlas_cloud_provider_access_setup](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cloud_provider_access#mongodbatlas_cloud_provider_access_setup) | resource |
+| [mongodbatlas_cloud_provider_access_authorization](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cloud_provider_access#mongodbatlas_cloud_provider_access_authorization) | resource |
+| [aws_iam_role](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_role) | resource |
+| [aws_iam_role_policy](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_role_policy) | resource |
 
-## Considerations 
+-> **NOTE:** Encryption at rest can only be enabled once per project.
```
Comment (AgustinBettati) [inline modules/aws-kms/README.md:43]: good call out, would make sure we add a similar note in push based logging

**Review by AgustinBettati — APPROVED** [review]
LGTM

**Review by kanchana-mongodb — APPROVED** [review]

**Pull request merged**